source/PlayServicesResolver/src/ResolverVer1_1.cs

```diff
@@ -848,12 +858,26 @@ public override int Version()
                 DoResolutionUnsafe(svcSupport, destinationDirectory,
                                    unlockResolveAndSignalResolutionComplete);
             };
-            lock (resolveLock) {
-                resolveUpdateQueue.Enqueue(resolve);
-                EditorApplication.update += UpdateTryResolution;
+
+            if (runSynchronously)
+            {
+                resolve();
+            }else
+            {
+                lock (resolveLock)
+                {
+                    resolveUpdateQueue.Enqueue(resolve);
+                    EditorApplication.update += UpdateTryResolution;
+                }
```

Comment:
There are a lot of code paths that register with EditorApplication.update to do stuff in future on the main thread.  This makes me wonder whether it would be better to instead write a method that wraps PlayServicesResolver.updateQueue and allows everything to optionally run synchronously ?  I'm not sure why resolveUpdateQueue is separate to PlayServicesResolver.updateQueue to be honest.

Reply:
Keep in mind that my patch was/is not intended to make the whole system support synchronous execution. I only focused on ResolverVer1_1, as that is the only part of this whole system we use. Making the whole system compatible is a task i feel is a bit too big for the changes we need.